pkg/asset/machines/aws/worker.go

```diff
@@ -53,12 +53,12 @@ items:
     replicas: {{$instance.Replicas}}
     selector:
       matchLabels:
-        sigs.k8s.io/cluster-api-machineset: worker
+        sigs.k8s.io/cluster-api-machineset: {{$c.ClusterName}}-worker-{{$index}}
```

Comment:
Do we really want to use the index here? This implies that the order of the worker machine pools in the install config is important, which I don't think that it is. Shouldn't we be using the name of the machine pool from the install config instead?

Reply:
Hey @staebler this is just being consistent with the machineset name https://github.com/openshift/installer/pull/519/files#diff-a416ed747e927b21b9c26916a1129cdfR46
To reduce possible side effects for machines matching more than one owner label
https://github.com/kubernetes-sigs/cluster-api/blob/master/pkg/controller/machineset/controller.go#L304

Reply:
I'm not sure what the purpose of having a name in the install-config.yml for each worker machine pool is if we are not going to use it for the name of the machine set.